app/results.py

```diff
@@ -188,8 +188,12 @@ def get(self):
             # If a query looks like a phone number, show the user a result
             # of looking up the number in the carriers-provided BBS system.
             if self.config.jp_mobile_carrier_redirect:
-                if jp_mobile_carriers.handle_phone_number(self, query.query):
-                    return 
+                try:
+                    if jp_mobile_carriers.handle_phone_number(self, query.query):
+                        return
+                except Exception as e:
+                    logging.error('failed to scrape phone number')
```

Comment:
> Do we need a message with i18 translation support in this case as we show the message to users? (I guess that _("a message") means in the example).

I18n the message shown to the end users using _("..."), but do not i18n the message for logging, which makes it hard to grep the log, for example.

> Should the return code be 500 (internal server error) in this case?

Yes.

> (Sorry for asking too many little details. I'm new to the codebase so please bear with me for now.)

No problem at all.